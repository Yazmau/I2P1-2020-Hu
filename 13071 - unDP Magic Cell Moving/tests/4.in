1000
1 0 10865589
4
2 0
2 2
1 0 377877641
2 0
3 386364087
2 2
3 946179272
2 0
2 2
4
1 2 278784499
3 415937285
2 0
2 2
2 0
4
4
4
3 662160481
2 2
2 0
3 300588468
3 296373539
3 590088239
2 2
4
3 597165840
2 0
2 2
1 0 46734131
2 2
2 0
1 2 67197385
4
4
1 0 887021755
1 0 429888561
4
2 2
4
4
1 2 578074934
3 394847727
2 0
2 0
4
1 0 553001291
1 0 767147747
2 2
3 862724604
1 2 407867578
4
3 817297238
4
3 699468109
2 0
3 472388559
2 2
2 2
4
3 245070481
3 916491364
1 2 488336028
3 918778385
1 2 814043528
3 858065473
1 0 512300980
2 2
1 0 703431349
4
2 0
1 0 236425686
1 0 999234891
1 0 651772445
3 578451425
2 2
3 749919926
3 841081686
3 76351558
3 478558052
2 0
4
4
3 224733194
4
1 0 351095108
3 617039081
4
1 2 502739669
3 476109634
3 264128329
2 2
3 488645891
3 176426987
3 449696472
1 0 991029916
2 0
2 0
4
4
4
1 0 535534798
4
3 606480861
2 0
2 0
4
3 377603307
2 2
2 2
3 118491786
2 2
3 777567754
4
4
4
1 2 121724322
4
2 2
4
1 0 95942252
1 2 465315985
2 2
1 2 752198818
4
2 2
2 0
1 0 409177247
1 0 990241274
4
1 0 341445395
1 0 378171268
2 0
2 2
2 0
1 0 402353453
1 0 330339817
3 416015676
4
1 0 171137267
3 605847425
1 0 216297721
1 2 236482519
3 570367458
1 0 571128415
2 2
3 721775367
3 951965392
4
3 335713388
1 2 106198913
2 0
4
3 367222171
3 4159246
4
1 2 814600010
3 91095740
3 745909249
2 2
2 2
1 2 197517458
1 2 817932701
3 970784208
1 0 576330668
1 0 591633079
1 2 345984056
4
4
3 846208552
1 2 479706621
1 0 169073176
4
2 2
1 2 981611713
3 753183000
1 0 427085063
2 0
2 0
3 891186182
4
2 2
2 2
3 902746763
3 374221929
1 2 467804613
2 2
4
2 0
2 0
2 0
2 2
2 2
4
3 418289866
3 992605925
4
2 2
3 449449411
4
4
3 65681883
4
2 0
3 62869555
4
1 0 412189456
3 757362592
1 2 212912144
1 2 210186140
2 2
2 0
2 2
1 2 56111314
2 0
1 2 68380665
3 238918802
3 35003289
3 469641012
2 0
2 2
2 0
4
2 0
2 0
1 0 72686267
1 0 106633018
3 52384970
2 2
4
3 3629930
2 2
2 0
1 2 791488232
2 0
2 0
2 0
2 2
2 0
3 521878346
4
1 2 256588541
3 213843836
2 0
2 0
2 2
1 2 526802929
4
1 0 207126517
4
3 152524011
3 1428501
4
2 0
4
3 122549422
1 0 794377450
1 0 588393203
4
3 645794845
1 0 434761967
2 0
4
1 0 851744006
2 0
2 2
3 607947916
2 0
3 863481956
3 501408904
1 2 56180998
2 2
1 0 766296972
1 0 344520743
2 2
4
3 568612435
4
3 487495207
2 2
4
1 0 730594880
1 0 887973313
1 2 65569211
1 2 936651533
1 2 510985611
4
1 2 169785210
4
2 2
4
4
2 2
4
4
2 2
4
3 276897684
1 0 857176628
4
2 2
3 352380318
2 0
4
2 2
3 327757962
1 2 960929254
1 0 336916201
3 493601049
4
1 0 4672763
3 508886424
4
2 0
2 0
4
1 2 957793911
3 432609200
4
4
2 0
4
3 891302454
2 2
1 2 468754336
4
3 231707837
2 2
1 0 806772484
4
2 2
2 2
2 2
3 50509435
2 0
4
2 2
4
2 0
4
4
4
2 0
4
4
4
3 889477081
4
2 0
3 173233330
2 2
4
2 2
4
2 2
4
2 0
2 0
3 764504043
3 180069688
2 2
3 384349374
4
4
4
4
2 2
4
2 0
3 584528446
4
2 0
1 2 155349605
1 0 562956979
2 2
1 2 15919859
4
3 708634308
4
2 2
3 442518256
4
1 2 837964641
1 0 114683283
4
1 0 289714488
1 2 360367181
3 613920607
3 731623809
1 0 202535405
3 439772658
2 0
4
4
1 2 897273713
2 2
4
3 594767041
3 644433895
3 4462889
1 0 937185711
3 160773768
4
1 0 978781114
4
4
3 108696242
2 0
2 0
1 0 301167792
1 2 973449846
2 2
3 984269622
4
3 44376017
4
3 654592167
2 0
3 431649508
2 0
1 2 227725021
1 0 306018723
3 83076912
3 568019890
3 673062298
4
2 2
4
2 0
3 893667177
4
2 2
2 0
1 2 90488650
2 2
4
2 0
4
2 2
4
1 0 980990562
1 2 19759755
4
3 685007303
4
4
2 0
2 2
1 2 845458799
4
2 0
3 410869281
2 2
2 2
2 2
2 2
4
4
2 2
2 0
3 554027873
4
1 0 520829467
1 0 814733857
4
1 0 815818781
1 0 672858021
3 667037133
1 0 861097445
1 0 839263984
3 135993785
2 2
2 0
4
4
3 295714283
4
4
1 0 422151116
2 2
2 0
4
2 0
3 105785688
1 2 919589389
2 2
3 820809687
2 2
4
1 0 121520914
3 235609472
4
1 2 694380532
2 2
3 573558699
3 499805958
4
2 0
4
4
1 2 955095788
2 0
1 0 343292651
2 2
2 2
3 430595883
2 0
4
1 2 879205568
1 0 923341692
4
1 2 879273312
2 0
3 373957855
3 88470201
2 0
1 2 925974978
4
1 0 851706579
1 2 403918892
4
3 249798136
1 2 945064988
3 720338129
2 0
1 0 618190006
3 199850193
4
1 0 394321150
1 2 996853294
2 0
1 0 930942823
3 840413674
3 482989127
2 2
4
1 2 16830954
1 0 91576819
3 32441155
3 911893091
1 0 471202434
1 2 423256592
4
4
1 2 998242376
3 36388346
3 340084998
2 0
1 2 788418649
1 0 301412870
3 831544474
4
4
3 112778458
2 0
3 590551964
2 0
1 2 106032053
1 0 826992179
2 2
2 0
4
2 0
1 2 398852608
2 2
2 0
4
4
4
2 2
4
1 2 793954522
3 853529180
4
2 0
3 276040424
1 0 246705713
1 2 500348025
3 700174585
2 2
4
3 96419302
1 0 638527176
3 721718073
3 806931084
4
4
4
3 557739385
1 2 556257220
1 2 165803708
4
3 395414807
1 0 854315821
4
4
3 244816875
2 2
1 0 377502107
2 0
3 435193775
2 0
2 2
2 0
3 917460815
2 0
1 0 566259082
1 0 223226089
3 27577548
4
2 0
4
4
1 0 588220595
3 196246769
2 0
1 2 894273448
3 205099060
3 624492460
3 588384556
4
3 670974418
3 121151220
4
4
3 415422672
4
4
1 2 344935036
2 0
3 688028723
1 0 291205115
1 0 398411934
3 592664575
3 572803428
1 2 525997450
1 2 760867507
1 0 139880940
1 2 314971999
4
2 0
2 2
2 2
1 2 889675235
3 497142356
4
1 2 654043954
4
3 706981889
3 802496510
1 0 819006862
4
4
3 296301780
1 0 942087829
3 263280980
2 0
2 0
3 665248630
3 436771642
4
4
2 2
2 0
1 0 546426945
2 0
4
1 2 700203504
4
1 0 971215028
2 0
1 2 659061965
1 0 83287117
4
3 511543894
1 0 131989707
2 2
4
1 0 516938871
3 92943144
4
4
4
1 0 982672817
4
1 2 780046621
3 955023475
3 960381274
1 0 456036017
2 0
1 2 590361971
1 2 558948798
3 244261650
4
3 190675580
2 2
3 253970832
4
2 2
1 0 684057684
3 462939390
2 2
1 0 672451925
1 2 823628803
3 419401935
2 0
1 2 145212650
4
1 0 558474021
3 646424858
3 247584084
1 2 943361056
3 800683747
3 552543780
3 319818325
1 2 914990725
4
2 0
1 0 719966726
4
1 0 763569135
2 2
2 2
1 2 588514542
1 2 110915662
2 2
2 2
2 2
4
2 2
4
3 53007430
1 0 941160460
4
2 2
3 48032688
2 2
1 0 921571571
2 0
2 0
3 628833654
3 339184899
4
3 946672421
1 0 236317730
4
4
2 0
3 715254746
2 0
1 2 352473365
4
2 0
2 0
3 17883237
4
3 116085101
3 617689549
2 2
2 0
4
2 0
1 2 839864300
4
1 2 154898633
1 2 253737725
4
4
3 568318440
2 2
4
2 2
4
3 218384074
2 0
3 552431086
3 748999465
2 0
2 0
2 2
4
4
1 2 713441454
2 2
4
3 440134876
1 2 836153595
1 2 191633458
2 2
2 0
3 232630576
1 2 537512116
4
2 0
3 662506469
1 2 504367804
3 382366058
1 2 604226147
3 179287196
1 0 166017447
1 0 74395611
1 2 639056121
3 396412131
4
1 2 736502159
3 27417913
4
2 0
2 0
3 346076456
4
2 0
2 2
2 0
3 447569217
4
1 2 65195222
3 262562955
4
3 987076390
2 2
1 0 625089507
2 0
3 145549018
2 2
2 2
3 542658413
1 0 746422423
4
4
2 0
1 2 352805766
1 2 41793228
1 0 505201864
3 140161340
4
4
2 2
2 2
3 71770894
4
3 122486008
1 0 498134477
3 858991129
2 2
4
4
3 95939292
4
1 2 663114674
3 647757716
2 0
4
2 2
1 2 792684469
2 2
1 0 889186799
4
4
3 406812118
2 0
1 0 220295072
2 0
1 2 466431750
1 2 692429554
2 0
4
2 2
2 2
3 123822912
2 2
3 98487112
3 975524775
3 304016287
2 2
3 238531930
1 0 152360517
2 0
3 518807865
4
2 0
2 2
3 510273942
4
1 0 354474569
1 2 407223076
3 656233348
3 286000780
2 2
3 269178951
2 2
4
1 2 815016567
2 2
4
4
1 2 763192612
1 2 490675253
1 0 542592872
4
2 0
4
2 2
2 2
2 0
1 2 562130920
4
1 0 964096927
2 0
1 0 53040824
3 877538098
3 291878728
2 2
3 110355902
4
1 0 117683015
1 0 572347466
3 139185332
3 450448064
4
4
3 24073306
4
1 0 738291565
2 0
3 65126918
4
1 0 260055925
3 785416932
1 0 439865173
2 2
2 2
2 0
1 0 580563089
2 0
1 0 369772238
2 2
1 2 910872652
2 0
3 385909856
4
3 893925231
1 2 13895090
1 2 918509604
2 2
2 0
3 360515835
3 984818441
2 0
3 245804413
1 0 398028117
4
2 2
1 2 201699886
2 0
3 241318888
2 2
4
1 2 250079181
2 2
3 951295493
2 2
3 872947597
1 0 148401821
2 2
3 432585370
2 2
3 396229294
2 0
2 0
2 0
4
4
4
3 274249321
2 0
4
1 2 918990059
4
3 74782661
2 2
3 98888034
1 2 34932830
1 0 215734126
4
1 0 957487286
4
4
3 333387633
3 242916447
3 791451414
1 2 31016491
4
3 507206121
4
2 2
4
4
