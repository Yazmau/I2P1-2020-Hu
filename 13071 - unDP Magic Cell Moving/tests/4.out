0
Invalid
946179272
386364087
386364087
386364087
415937285
Invalid
Invalid
597165840
597165840
Invalid
Invalid
597165840
278784499
278784499
590088239
Invalid
Invalid
862724604
817297238
394847727
Invalid
Invalid
Invalid
Invalid
578074934
Invalid
Invalid
Invalid
578451425
578451425
224733194
Invalid
617039081
Invalid
Invalid
699468109
699468109
699468109
Invalid
699468109
553001291
777567754
777567754
777567754
Invalid
777567754
449696472
Invalid
465315985
Invalid
Invalid
465315985
Invalid
Invalid
Invalid
Invalid
416015676
Invalid
Invalid
Invalid
Invalid
951965392
Invalid
570367458
4159246
Invalid
Invalid
Invalid
Invalid
Invalid
970784208
970784208
Invalid
Invalid
846208552
Invalid
891186182
Invalid
981611713
374221929
992605925
449449411
449449411
65681883
62869555
Invalid
Invalid
Invalid
Invalid
846208552
Invalid
Invalid
891186182
Invalid
521878346
Invalid
Invalid
472388559
Invalid
472388559
1428501
377603307
Invalid
122549422
Invalid
794377450
Invalid
Invalid
Invalid
645794845
568612435
1428501
Invalid
Invalid
Invalid
Invalid
Invalid
1428501
Invalid
1428501
213843836
213843836
777567754
777567754
52384970
Invalid
276897684
276897684
Invalid
Invalid
493601049
Invalid
508886424
777567754
Invalid
432609200
432609200
213843836
Invalid
432609200
Invalid
276897684
846208552
50509435
846208552
846208552
846208552
508886424
508886424
508886424
889477081
889477081
846208552
50509435
384349374
384349374
384349374
384349374
50509435
584528446
Invalid
Invalid
Invalid
584528446
708634308
442518256
Invalid
Invalid
442518256
Invalid
Invalid
Invalid
708634308
708634308
Invalid
439772658
Invalid
160773768
Invalid
160773768
160773768
Invalid
Invalid
984269622
44376017
Invalid
Invalid
673062298
431649508
893667177
Invalid
431649508
893667177
431649508
Invalid
Invalid
431649508
685007303
685007303
Invalid
685007303
65681883
65681883
554027873
Invalid
Invalid
554027873
Invalid
Invalid
Invalid
Invalid
135993785
135993785
295714283
295714283
Invalid
295714283
Invalid
56111314
Invalid
235609472
499805958
235609472
235609472
Invalid
Invalid
235609472
Invalid
Invalid
235609472
Invalid
Invalid
105785688
Invalid
Invalid
105785688
Invalid
Invalid
199850193
Invalid
Invalid
Invalid
199850193
Invalid
Invalid
Invalid
Invalid
911893091
911893091
Invalid
Invalid
Invalid
831544474
831544474
Invalid
Invalid
231707837
Invalid
891302454
891302454
891302454
231707837
Invalid
853529180
Invalid
Invalid
853529180
Invalid
806931084
806931084
806931084
Invalid
Invalid
557739385
Invalid
395414807
395414807
Invalid
Invalid
Invalid
27577548
487495207
487495207
Invalid
Invalid
588384556
121151220
121151220
415422672
415422672
Invalid
Invalid
Invalid
Invalid
Invalid
Invalid
572803428
Invalid
497142356
Invalid
497142356
Invalid
802496510
802496510
Invalid
436771642
436771642
546426945
Invalid
546426945
Invalid
971215028
Invalid
546426945
Invalid
92943144
92943144
92943144
Invalid
92943144
Invalid
Invalid
Invalid
Invalid
244261650
253970832
Invalid
Invalid
Invalid
Invalid
462939390
Invalid
Invalid
Invalid
319818325
Invalid
253970832
Invalid
Invalid
Invalid
27577548
917460815
Invalid
53007430
Invalid
339184899
Invalid
946672421
946672421
Invalid
196246769
17883237
617689549
Invalid
319818325
Invalid
Invalid
319818325
319818325
617689549
263280980
568318440
568318440
Invalid
748999465
Invalid
Invalid
Invalid
232630576
Invalid
Invalid
Invalid
Invalid
Invalid
396412131
Invalid
27417913
346076456
447569217
Invalid
262562955
Invalid
Invalid
542658413
542658413
Invalid
Invalid
Invalid
140161340
140161340
71770894
Invalid
232630576
232630576
95939292
Invalid
858991129
Invalid
Invalid
218384074
218384074
Invalid
Invalid
Invalid
542658413
Invalid
518807865
510273942
Invalid
Invalid
715254746
Invalid
946672421
946672421
Invalid
Invalid
Invalid
946672421
715254746
Invalid
946672421
Invalid
Invalid
110355902
Invalid
Invalid
450448064
450448064
24073306
Invalid
65126918
Invalid
Invalid
Invalid
Invalid
Invalid
385909856
Invalid
Invalid
Invalid
245804413
Invalid
984818441
Invalid
Invalid
951295493
951295493
951295493
984818441
Invalid
984818441
Invalid
Invalid
98888034
Invalid
98888034
98888034
Invalid
791451414
507206121
872947597
872947597
